react-leaflet/index.d.ts

```diff
@@ -0,0 +1,258 @@
+}
+
+
+interface PaneProps {
+    name?: string;
+    style?: React.CSSProperties;
+    className?: string;
+}
+interface Pane extends React.ComponentClass<PaneProps> { }
+declare const Pane: Pane;
+
+
+//There is no Layer class, these are the base props for all layers on the map
+interface LayerProps {
+    onadd?: (event: Leaflet.Event) => void;
+    onremove?: (event: Leaflet.Event) => void;
+
+    //Popup events
+    onpopupopen?: (event: Leaflet.PopupEvent) => void;
+    onpopupclose?: (event: Leaflet.PopupEvent) => void;
+
+    //Tooltip events
+    ontooltipopen?: (event: Leaflet.TooltipEvent) => void;
+    ontooltipclose?: (event: Leaflet.TooltipEvent) => void;
+}
+
+
+interface MarkerProps extends LayerProps, LeafletInteractionEvents, LeafletDraggingEvents {
+    position: Leaflet.LatLngExpression;
+    draggable?: boolean;
+    icon?: Leaflet.Icon;
+    zIndexOffset?: number;
+    opacity?: number;
+}
+interface Marker extends React.ComponentClass<MarkerProps> { }
+declare const Marker: Marker;
+interface MarkerInstance extends React.Component<MarkerProps, {}> {
+    leafletElement: Leaflet.Marker;
+}
+
+interface PopupProps extends LayerProps, Leaflet.PopupOptions {
+    position?: Leaflet.LatLngExpression;
+}
+interface Popup extends React.ComponentClass<PopupProps> { }
+declare const Popup: Popup;
+
+interface TooltipProps extends LayerProps, Leaflet.TooltipOptions { }
+interface Tooltip extends React.ComponentClass<TooltipProps> { }
+declare const Tooltip: Tooltip;
+
+interface GridLayerProps extends LayerProps {
+    opacity?: number;
+    zIndex?: number;
+
+    onloading?: (event: Leaflet.Event) => void;
+    ontileunload?: (event: Leaflet.TileEvent) => void;
+    ontileloadstart?: (event: Leaflet.TileEvent) => void;
+    ontileerror?: (event: Leaflet.TileErrorEvent) => void;
+    ontileload?: (event: Leaflet.TileEvent) => void;
+    onload?: (event: Leaflet.Event) => void;
+}
+interface GridLayer extends React.ComponentClass<GridLayerProps> { }
+declare const GridLayer: GridLayer;
+
+interface TileLayerProps extends GridLayerProps, Leaflet.TileLayerOptions {
+    url: string;
+}
+interface TileLayer extends React.ComponentClass<TileLayerProps> { }
+declare const TileLayer: TileLayer;
+
+interface ImageOverlayProps extends LayerProps, LeafletInteractionEvents {
+    url: string;
+    opacity?: string;
+}
+interface ImageOverlay extends React.ComponentClass<ImageOverlayProps> { }
+declare const ImageOverlay: ImageOverlay;
+
+interface WMSTileLayerProps extends TileLayerProps {
+    url: string;
+}
+interface WMSTileLayer extends React.ComponentClass<WMSTileLayerProps> { }
+declare const WMSTileLayer: WMSTileLayer;
+
+//Path is an abstract class
+interface PathProps extends LeafletLayerEvents, LeafletInteractionEvents, Leaflet.PathOptions {
+}
+
+
+interface CircleProps extends PathProps {
+    center: Leaflet.LatLngExpression;
+    radius?: number;
+}
+interface Circle extends React.ComponentClass<CircleProps> { }
```

Comment:
I had started on this and reverted it as I was running in to trouble with `no-empty-interface`.
Have re-done it now and used a few `tslint:disable-next-line:no-empty-interface` as I'm not sure of a better way to do some of these.
The big one is `PathProps`, which saves me putting those 3 interfaces on the 5 `*Props` interfaces below.
If you know of a nicer way to do this I'd gladly change it. Thanks :)